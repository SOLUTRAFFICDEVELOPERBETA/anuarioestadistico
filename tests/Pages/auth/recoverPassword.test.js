import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { RouterContext } from 'next/dist/next-server/lib/router-context'
import RecoverPassword from '../../../pages/auth/recoverPassword';
import AuthContext from '../../../contexts/auth';
import AlertContext from '../../../contexts/alert';
import ThemeContext from '../../../contexts/theme';
import { act } from '@testing-library/react';

describe('Pruebas en <RecoverPassword/>', () => {
    const showMessage = jest.fn();
    const passwordReset = jest.fn();
    const bg = {
        name: 'imagen prueba',
        path: '/img/prueba.png',
        url: 'https://www.valladolid.mx/assets/img/default.jpg'
    }
    const router = {
        pathname: '/',
        route: '',
        basePatch: '',
        query: {},
        asPath: '',
        push: jest.fn(),
        replace: jest.fn(),
        reload: jest.fn(),
        back: jest.fn(),
        prefetch: jest.fn(),
        beforePopState: jest.fn(),
        listen: jest.fn(),
        isFallback: false,
        events: {
            on: jest.fn(),
            off: jest.fn(),
            emit: jest.fn(),
        },
    }
    const wrapper = mount(
        <RouterContext.Provider value={router}>
            <AlertContext.Provider value={{
                showMessage
            }}>
                <AuthContext.Provider value={{
                    passwordReset
                }}>
                    <ThemeContext.Provider value={{
                        bg
                    }}>
                        <RecoverPassword />
                    </ThemeContext.Provider>
                </AuthContext.Provider>
            </AlertContext.Provider>
        </RouterContext.Provider>
    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('No de bebe llamar la función passwordReset()', ()=> {
        wrapper.find('form').prop('onSubmit')({ preventDefault() { } });

        expect(passwordReset).not.toHaveBeenCalled();
        expect(passwordReset).toHaveBeenCalledTimes(0);
    })

    test('debe llamar la función passwordReset() y redireccionar  /auth', () => {
        const email = 'tuberquia2115@gmail.com'
        wrapper.find('input').simulate('change', { target: { value: email } });

        act(() => {
            wrapper.find('form').prop('onSubmit')({ preventDefault() { } });
        })

        expect(passwordReset).toHaveBeenCalled();
        expect(passwordReset).toHaveBeenCalledTimes(1);
        expect(router.push).toHaveBeenCalledWith('/auth');
    })
})