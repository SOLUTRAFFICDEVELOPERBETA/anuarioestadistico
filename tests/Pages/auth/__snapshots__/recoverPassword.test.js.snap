// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RecoverPassword/> debe mostrarse correctamente 1`] = `
<div
  className="css-6ctbyo-ContainerDiv e15g2ioh0"
>
  <div
    className="css-1m18txs-AccountCard e1ghh5810"
  >
    <div
      className="css-rgn295-AccountHead e1ghh5812"
    >
      <h3>
        <span
          className="accountlogoaccent"
        >
          Recuperación de Contraseña
        </span>
      </h3>
    </div>
    <form
      className="css-rwem1p-Form e1ghh5811"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
        >
          Correo Electrónico
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Correo Electrónico de recuperación."
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Correo Electrónico
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiButtonGroup-root MuiButtonGroup-fullWidth MuiButtonGroup-vertical"
        role="group"
        style={
          Object {
            "margin": "10px 0",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlined MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Enviar Correo
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlined MuiButton-textPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Regresar al inicio
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
